ProgressBarState.String() panics on out-of-range values

**Describe the bug**

Calling `ProgressBarState.String()` appears to panic when the receiver value is outside the `[0, 4]` range.

The method looks like this:

```go
func (s ProgressBarState) String() string {
    return [...]string{
        "None",
        "Default",
        "Error",
        "Indeterminate",
        "Warning",
    }[s]
}
```

`ProgressBarState` is defined as `type ProgressBarState int`, and only five constants (0–4) are defined. The array literal is fixed at length 5 and indexed directly with `s`, so a value outside that range might cause an index-out-of-range panic.

`ProgressBar.State` is also an exported field, so it might be possible to assign arbitrary values without going through the `NewProgressBar` constructor.

---

**Setup**

- OS: not terminal-specific
- Shell: not terminal-specific
- Terminal Emulator: not terminal-specific
- Terminal Multiplexer: not terminal-specific

The issue appears to be in pure Go runtime code, unrelated to terminal environment.

---

**To Reproduce**

1. Create a file `main.go` with the following content:

```go
package main

import tea "charm.land/bubbletea/v2"

func main() {
    var s tea.ProgressBarState = 5
    println(s.String())
}
```

2. Run `go run main.go`

3. Observe the output:

```
panic: runtime error: index out of range [5] with length 5
```

Negative values seem to behave similarly — replacing `5` with `-1` produces:

```
panic: runtime error: index out of range [-1]
```

---

**Source Code**

```go
// tea.go, around line 321
func (s ProgressBarState) String() string {
    return [...]string{
        "None",
        "Default",
        "Error",
        "Indeterminate",
        "Warning",
    }[s]
}
```

---

**Expected behavior**

Unsure if this is intentional. The method might simply never be called with unexpected values in practice, in which case this is a non-issue. If out-of-range inputs are a realistic concern, some bounds protection might be needed so the method doesn't panic.

---

**Screenshots**

N/A

---

**Additional context**

N/A

--- a/tea.go
+++ b/tea.go
@@ -317,7 +317,8 @@ const (
 	ProgressBarWarning
 )
 
-// String return a human-readable value for the given [ProgressBarState].
+// String returns a human-readable name for the given [ProgressBarState].
+// Values outside the known range return "Unknown".
 func (s ProgressBarState) String() string {
 	states := [...]string{
 		"None",

--- a/tea.go
+++ b/tea.go
@@ -320,17 +320,20 @@ const (
 // String returns a human-readable name for the given [ProgressBarState].
 // Values outside the known range return "Unknown".
 func (s ProgressBarState) String() string {
-	states := [...]string{
-		"None",
-		"Default",
-		"Error",
-		"Indeterminate",
-		"Warning",
-	}
-	if s < 0 || int(s) >= len(states) {
+	switch s {
+	case ProgressBarNone:
+		return "None"
+	case ProgressBarDefault:
+		return "Default"
+	case ProgressBarError:
+		return "Error"
+	case ProgressBarIndeterminate:
+		return "Indeterminate"
+	case ProgressBarWarning:
+		return "Warning"
+	default:
 		return "Unknown"
 	}
-	return states[s]
 }
 
 // ProgressBar represents the terminal progress bar.

--- a/tea.go
+++ b/tea.go
@@ -319,13 +319,17 @@ const (
 
 // String return a human-readable value for the given [ProgressBarState].
 func (s ProgressBarState) String() string {
-	return [...]string{
+	states := [...]string{
 		"None",
 		"Default",
 		"Error",
 		"Indeterminate",
 		"Warning",
-	}[s]
+	}
+	if s < 0 || int(s) >= len(states) {
+		return "Unknown"
+	}
+	return states[s]
 }
 
 // ProgressBar represents the terminal progress bar.

--- a/tea_test.go
+++ b/tea_test.go
@@ -669,3 +669,23 @@ func BenchmarkTeaRun(b *testing.B) {
 		_ = r.CloseWithError(io.EOF)
 	}
 }
+
+// TestProgressBarStateStringOutOfRange is a regression test for
+// https://github.com/charmbracelet/bubbletea/issues/1711: String panicked
+// with an index-out-of-range error for any ProgressBarState outside the
+// [ProgressBarNone, ProgressBarWarning] range. Since State is an exported
+// field on ProgressBar, callers can assign such a value without going
+// through NewProgressBar.
+func TestProgressBarStateStringOutOfRange(t *testing.T) {
+	for _, s := range []ProgressBarState{-1, 5, 100} {
+		if got, want := s.String(), "Unknown"; got != want {
+			t.Errorf("ProgressBarState(%d).String() = %q, want %q", int(s), got, want)
+		}
+	}
+
+	for s := ProgressBarNone; s <= ProgressBarWarning; s++ {
+		if got := s.String(); got == "Unknown" {
+			t.Errorf("ProgressBarState(%d).String() = %q, want a known name", int(s), got)
+		}
+	}
+}

